@xen-orchestra/web/src/components/backups/backed-up-vms/backedUpVmsRuns.vue

```diff
@@ -0,0 +1,184 @@
+        </tr>
+      </template>
+    </VtsDataTable>
+    <UiTopBottomTable :selected-items="0" :total-items="0">
+      <UiTablePagination v-bind="paginationBindings" />
+    </UiTopBottomTable>
+  </UiCard>
+</template>
+
+<script setup lang="ts">
+import { useXoBackupLogsUtils } from '@/composables/xo-backup-log-utils.composable'
+import { useXoVmCollection } from '@/remote-resources/use-xo-vm-collection'
+import type { XoBackupLog } from '@/types/xo/backup-log.type'
+import type { VmsSmartModeDisabled, VmsSmartModeEnabled, XoVmBackupJob } from '@/types/xo/vm-backup-job.type'
+import { VM_POWER_STATE } from '@/types/xo/vm.type.ts'
+import { extractIdsFromSimplePattern } from '@/utils/pattern.util'
+import type { IconName } from '@core/icons'
+import type { Branded } from '@core/types/utility.type'
+import VtsDataTable from '@core/components/data-table/VtsDataTable.vue'
+import VtsIcon from '@core/components/icon/VtsIcon.vue'
+import UiCard from '@core/components/ui/card/UiCard.vue'
+import UiLink from '@core/components/ui/link/UiLink.vue'
+import UiQuerySearchBar from '@core/components/ui/query-search-bar/UiQuerySearchBar.vue'
+import UiTablePagination from '@core/components/ui/table-pagination/UiTablePagination.vue'
+import UiTitle from '@core/components/ui/title/UiTitle.vue'
+import UiTopBottomTable from '@core/components/ui/top-bottom-table/UiTopBottomTable.vue'
+import { usePagination } from '@core/composables/pagination.composable'
+import { useRouteQuery } from '@core/composables/route-query.composable'
+import { useTable } from '@core/composables/table.composable'
+
+import { formatSizeRaw } from '@core/utils/size.util'
+import * as ValueMatcher from 'value-matcher'
+import { computed, ref } from 'vue'
+import { useI18n } from 'vue-i18n'
+
+const { backedUpVms: rawBackedUpVms, backupLogs } = defineProps<{
+  backedUpVms: XoVmBackupJob['vms']
+  backupLogs: XoBackupLog[]
+}>()
+
+const { t } = useI18n()
+
+const searchQuery = ref('')
+const { getVmsByIds, vms } = useXoVmCollection()
+const { findTransferTask } = useXoBackupLogsUtils()
+const selectedBackupJobId = useRouteQuery('id')
+
+function checkSmartModeEnabled(
+  value: VmsSmartModeEnabled | VmsSmartModeDisabled | undefined
+): value is VmsSmartModeEnabled {
+  if (value === undefined || typeof value !== 'object' || value === null) {
+    return false
+  }
+
+  return !('id' in value)
+}
+
+const backedUpVms = computed(() => {
+  let filteredVms = []
+
+  if (checkSmartModeEnabled(rawBackedUpVms)) {
+    const predicate = ValueMatcher.createPredicate(rawBackedUpVms)
+    filteredVms = vms.value.filter(vm => predicate(vm) && !vm.tags?.includes('xo:no-bak'))
+  } else {
+    filteredVms = getVmsByIds(extractIdsFromSimplePattern(rawBackedUpVms) as Branded<'vm'>[])
+  }
+
+  if (searchQuery.value) {
+    const query = searchQuery.value.toLowerCase()
+    filteredVms = filteredVms.filter(vm => vm.name_label.toLowerCase().includes(query))
+  }
+
+  return filteredVms
+})
```

Comment:
```suggestion
const backedUpVms = computed(() => {
  if (checkSmartModeEnabled(rawBackedUpVms)) {
    const predicate = ValueMatcher.createPredicate(rawBackedUpVms)

    return vms.value.filter(vm => predicate(vm) && !vm.tags?.includes('xo:no-bak'))
  }
  return getVmsByIds(extractIdsFromSimplePattern(rawBackedUpVms) as Branded<'vm'>[])
})

const filteredBackedUpVms = computed(() => {
  const searchTerm = searchQuery.value.trim().toLocaleLowerCase()

  if (!searchTerm) {
    return backedUpVms.value
  }

  return backedUpVms.value.filter(backupJob =>
    Object.values(backupJob).some(value => String(value).toLocaleLowerCase().includes(searchTerm))
  )
})
```

I think we can do something like that to have a better readability